### ISSUE TITLE
When lossing focus, context gets lost

### ISSUE THREAD

**Issue by amenk** [bitwarden/browser#443]
1. Open Bitwarden dropdown 
2. Click for example add account
3. Do something in the browser window or anywhere else
4. Popup closes
5. When you open it, you are still in the main view

Two things to fix:

1. Popup should keep sticky (not sure if it is possible)?
2. If it is not possible, when reopening, you should be on the same page like before at least

**Comment by kspearrin** [comment]
This is not possible since the browser controls when that window closes. This is why we offer the ability to "pop out" to a new window with the arrow in the top left corner.

**Comment by amenk** [comment]
Okay, but it would be possible that the popup content is preserved, isn't
it?​

**Comment by kspearrin** [comment]
That would require a lot of work to try and preserve state between open/close. All state is lost when the popup closes each time so we would have to somehow store that and reinstate it each time you open.

**Comment by amenk** [comment]
​Hmm .. currently it's quite an UX problem. I will try to remember by
myself to pop-out the window though ..

**Comment by kustodian** [comment]
This is for me the biggest UX frustration.

Would it be possible to make Bitwarden open in a sidebar when clicking on the Bitwarden icon in the toolbar, instead of opening it in the popup? That would solve the closing, since you would close only when you save? I'm not a fan of popups, because they are a new window and remembering to open the popup is also bad.

**Comment by kspearrin** [comment]
@kustodian No, the browser controls when happens when you click that icon and that is to always open the popup.

Chrome does not have a sidebar. Only FF and Opera.

**Comment by GeoDosch** [comment]
I just wanted to add my support for this issue.  After years of using LastPass (and RoboForm before that), I'm a huge fan of Bitwarden.  However, this behavior has tripped me up more times than I like to admit: I'm on a site where I need to add my credentials to BW (since it hasn't offered to save the site for me.)  I click the icon, and then click the "Add a Login" link (that thing really sucks you in!)  I'll start making my edits to the site name, user name, URI.  Then I get to the Password field, so I switch over to the window where I've temporarily saved that, and BLAM!, the BW window is gone.  Then I curse myself for yet again forgetting that this was going to happen, and start all over with the BW screen in the sidebar.

It's happened to me enough that I know I can't be the only one who keeps forgetting about that 'feature'.  And since I realize what's occurring, I know it's not an easy fix.  But one thing that might help is if on that screen where it says "Add a Login", there was a reminder to open BW in a more persistent mode before attempting that.  The pop-out window never works for me on FF (I know there was an issue here somewhere about that) so I opt for the sidebar.  In Chrome, which I rarely use, the pop-out seems to work okay.  Either way, a reminder to use one of those for saving a new site would go a long way towards also saving my sanity.  :)

**Comment by amenk** [comment]
When trying to add an attachment to an entry, this is not working at all currently - the file chooser closes the popup.

My suggestion: Just use pop-out form automatically when editing entries.

**Comment by kspearrin** [comment]
@amenk This is a firefox bug that has been reported for quite a while.

**Comment by amenk** [comment]
I use chrome...

**Comment by kspearrin** [comment]
@amenk I just tested Chrome on Windows by choosing a file under the attachments edit page. The popup does not close. What OS are you on?

**Comment by amenk** [comment]
Ubuntu with xfce
